src/System.Net.Http/src/System/Net/Http/SocketsHttpHandler/Http2Connection.cs

```diff
@@ -830,6 +830,11 @@ private async ValueTask<Http2Stream> SendHeadersAsync(HttpRequestMessage request
 
                 while (remaining.Length > 0)
                 {
+                    if (cancellationToken.IsCancellationRequested)
+                    {
+                        throw new OperationCanceledException(cancellationToken);
```

Comment:
I wonder if it's actually legal to send RST_STREAM in the middle of headers (i.e. after one or more HEADERS/CONTINUATION frames but before the EndHeaders flag is sent). What does the spec say here?

Reply:
Ah, that's a reasonable behavior. I'll remove this.

Reply:
This does complicate the question of cancellation and how it interacts with FlushOutgoingBytesAsync, because it means we either need to cancel before we send HEADERS at all, or we need to send the full HEADERS and CONTINUATION blocks, while still allowing the request to be cancelled while the send of those is in progress...